src/wallet/wallet.cpp

```diff
@@ -4502,34 +4484,40 @@ bool CWallet::IsLockedCoin(uint256 hash, unsigned int n) const
     return (setLockedCoins.count(outpt) > 0);
 }
 
-void CWallet::ListLockedCoins(std::vector<COutPoint>& vOutpts) const
+std::vector<COutPoint> CWallet::ListLockedCoins() const
 {
     AssertLockHeld(cs_wallet);
-    for (std::set<COutPoint>::iterator it = setLockedCoins.begin();
-         it != setLockedCoins.end(); it++) {
-        COutPoint outpt = (*it);
-        vOutpts.push_back(outpt);
+    std::vector<COutPoint> ret;
+    for (const auto& output : setLockedCoins) {
+        ret.push_back(output);
     }
+    return ret;
 }
 
-void CWallet::ListProTxCoins(std::vector<COutPoint>& vOutpts) const
+std::vector<COutPoint> CWallet::ListProTxCoins() const { return ListProTxCoins(setWalletUTXO); }
+
+std::vector<COutPoint> CWallet::ListProTxCoins(const std::set<COutPoint>& utxos) const
 {
-    // TODO: refactor duplicated code between CWallet::AutoLockMasternodeCollaterals and CWallet::ListProTxCoins
-    if (!m_chain) {
-        vOutpts.clear();
-        return;
+    AssertLockHeld(cs_wallet);
+
+    if (!m_chain) return std::vector<COutPoint>();
+
+    std::vector<std::pair<const CTransactionRef&, unsigned int>> candidates;
```

Comment:
what does this unsigned int represent? add comments please

Reply:
Changed to `uint32_t` to match `COutPoint::n` type ([source](https://github.com/dashpay/dash/blob/a4e6d1100f382a0005a9be643e9314bedc80c1c9/src/primitives/transaction.h#L34)), commented as `/*index=*/`, resolved in latest push.